tests/acceptance/30_generic_methods/schedule_simple.cf

```diff
@@ -0,0 +1,129 @@
+
+  reports:
+    ok::
+      "$(this.promise_filename) Pass";
+    !ok::
+      "$(this.promise_filename) FAIL";
+}
+
+bundle agent schedule_test(test_id, test_now, test_splay, agent_periodicity, delay_minutes, delay_hours,
+                           start_on_minutes, start_on_hours, start_on_day_of_week,
+                           periodicity_minutes, periodicity_hours, periodicity_days, mode,
+                           exp_current_run, exp_run, exp_error)
+{
+  vars:
+    any::
+      "date" string => execresult("date --date='${test_now}' +%s", "useshell");
+
+  classes:
+    pass2::
+      "ok_run" expression => strcmp("${exp_current_run}", "${schedule_generic.current_run_id}");
+      "ok_job" expression => strcmp("${exp_run}", "${schedule_generic.job_run_id}");
+      "expect_error" expression => strcmp("${exp_error}", "ko");
+    pass2.expect_error::
+      "ok_error" expression => "job_${test_id}_error";
+    pass2.!expect_error::
+      "ok_error" expression => "!job_${test_id}_error";
+
+    pass2::
+      "global_error" expression => "(expect_error.!ok_error)|(!expect_error.(!ok_run|!ok_job|!ok_error))",
+                     scope => "namespace";
+
+    any::
+      "pass3" expression => "pass2";
+      "pass2" expression => "any";
+
+  methods:
+    "date"  usebundle => ncf_date("${date}");
+    "splay" usebundle => ncf_splay("${test_splay}");
+    "test"  usebundle => schedule_simple("${test_id}", "${agent_periodicity}", "${delay_minutes}", "${delay_hours}",
+                                         "${start_on_minutes}", "${start_on_hours}", "${start_on_day_of_week}",
+                                         "${periodicity_minutes}", "${periodicity_hours}", "${periodicity_days}", "${mode}");
+
+  reports:
+#    any::
+#      "-- test -- ${test_id}";
+#      " prefix ${schedule_generic.test}";
+#      " timestamp = ${date.timestamp}";
+#      " splay_dec = ${splay.splay_dec}";
+#      " run_shift = ${schedule_generic.run_shift}";
+#      " max_run_id = ${schedule_generic.max_run_id}";
+#      " run_id = ${schedule_generic.job_run_id}";
+#      " elapsed = ${schedule_generic.elapsed}";
+#      " current_run_id = ${schedule_generic.current_run_id}";
+#
+#      " splay = ${schedule_simple_generic.splay}";
+#      " periodicity = ${schedule_simple_generic.job_periodicity}";
+#      " now = ${schedule_simple_generic.now}";
+#
+#      " error_null_agent_periodicity" ifvarclass => "job_${test_id}_error_null_agent_periodicity";
+#      " error_interval_too_big" ifvarclass => "job_${test_id}_error_splay_too_big";
+#      " error_interval_too_small" ifvarclass => "job_${test_id}_error_splay_too_small";
+#      " Error !" ifvarclass => "job_{test_id}_error";
+#      " Must RUN !" ifvarclass => "job_${test_id}_run";
+#      " After run ${test_id}!" ifvarclass => "job_${test_id}_after_run";
+#      " After splay ${test_id}!" ifvarclass => "job_${test_id}_after_splay";
+
+    pass3.expect_error.ok_error::
+      "TEST result OK ${test_id}";
```

Comment:
i really fail to see what is being tested here :/ could you explain a bit more in the code please ?